core/chains/evm/logpoller/log_poller.go

```diff
@@ -679,9 +679,7 @@ func (lp *logPoller) backfill(ctx context.Context, start, end int64) error {
 		}
 
 		lp.lggr.Debugw("Backfill found logs", "from", from, "to", to, "logs", len(gethLogs), "blocks", blocks)
-		err = lp.orm.Q().WithOpts(pg.WithParentCtx(ctx)).Transaction(func(tx pg.Queryer) error {
-			return lp.orm.InsertLogs(convertLogs(gethLogs, blocks, lp.lggr, lp.ec.ConfiguredChainID()), pg.WithQueryer(tx))
-		})
+		err = lp.orm.InsertLogs(convertLogs(gethLogs, blocks, lp.lggr, lp.ec.ConfiguredChainID()), pg.WithParentCtx(ctx))
```

Comment:
This is a single query, we don't have to open TX for that

Reply:
Weird, not sure how it got like that!

Reply:
If we remove this outer tx, we need to open a tx inside InsertLogs since we may break that into multiple inserts

Reply:
Oh! Yes, seems like that should have always been down in the ORM. Maybe at some point there was another ORM call inside the same tx?

Reply:
This is my fault; I've removed that TX prematurely. We can just open TX within the insertLogs to guarantee they are inserted together.

Reply:
Kudos to @connorwstein for this finding!